### PR TITLE
Create curriculum pathway view

### PR DESCRIPTION
**Pull request by kelbyhawn**
Creates a `curriculum_pathway.haml` view for use throughout the code.org site. 
- Will be used on the new https://code.org/administrators page: https://github.com/code-dot-org/code-dot-org/pull/55895
- Fully responsive and accessible
- Works with LTR and RTL languages

**Jira ticket:** [ACQ-1405](https://codedotorg.atlassian.net/browse/ACQ-1405)

----

## Desktop

https://github.com/code-dot-org/code-dot-org/assets/9256643/3fffc935-0139-4310-9170-d29c804a65b8

### Keyboard Nav

https://github.com/code-dot-org/code-dot-org/assets/9256643/7718b5be-9845-4384-bb79-9f5cdd835e67

## Responsive

https://github.com/code-dot-org/code-dot-org/assets/9256643/e5db4463-e370-4d93-b5d6-0e61582f5627

## Mobile
Screenshot 2024-01-25 at 12 43 33 PM

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/curriculum_pathway.haml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      %p=hoc_s(:course_name_csp)
+    %a.course.csa.border-top-primary-default{href: "/csa"}
+      %p=hoc_s(:course_name_csa)
+
+  .background
+    - (1..13).each do |i|
+      %div
+
+%figure.curriculum-pathway.mobile
+  .school-wrapper
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/curriculum_pathway.haml:34]: I tried using helper methods for all of these but kept getting syntax errors — going with the long version in an effort to get this out 🥲

**Review by TurnerRiley — APPROVED** [review]
Wow Kelby this looks amazing !

**Comment by breville** [comment]
Very nice!  One suggestion is to render the ` >` transparently when not hovering, so that when hovering, the layout doesn't change.

**Comment by kelbyhawn** [comment]
> Very nice! One suggestion is to render the ` >` transparently when not hovering, so that when hovering, the layout doesn't change.

@breville the layout change was in the Figma mock, but I'll ask Moshe if he wants me to change it.

**Comment by breville** [comment]
> > Very nice! One suggestion is to render the ` >` transparently when not hovering, so that when hovering, the layout doesn't change.
> 
> @breville the layout change was in the Figma mock, but I'll ask Moshe if he wants me to change it.

Fair point.  I do thing it's a bit abrupt (text moving just because you hover over it feels less then ideal), and it does lead to some even more abrupt transitions at certain widths, like here:

https://github.com/code-dot-org/code-dot-org/assets/2205926/f85c0e53-9483-41d8-a5b4-6764e95f1977

**Comment by kelbyhawn** [comment]
> Fair point. I do thing it's a bit abrupt (text moving just because you hover over it feels less then ideal), and it does lead to some even more abrupt transitions at certain widths, like here…

I talked to Moshe and we have a solution: [Jira ticket](https://codedotorg.atlassian.net/browse/ACQ-1441)!

**Pull request merged**